### PR TITLE
Update data concepts docs for 2.0

### PR DESCRIPTION
**Pull request by PFedak**
Main changes:
 - swap put file default behavior
 - start using commit and sub-commit where appropriate
    (de-emphasizing sub-commit except where necessary)
 - add examples for commit-level commands
 - remove history option for list file and corresponding docs
    (code change should probably be in a separate commit I guess)
 - remove reference to branches not having heads

### CODE REVIEW

File: doc/docs/master/concepts/data-concepts/repo.md [hunk 1]
```diff
@@ -37,25 +37,29 @@ Pachyderm's repositories are divided into two categories:
     User repositories keep track of your data one commit at a time. 
     They further split into:
 
-    - **Input** repositories
+    - **Source** repositories
```
Comment (PFedak) [inline doc/docs/master/concepts/data-concepts/repo.md:40]: someone mentioned this renaming at some point, the idea being that even an "output" repo can be input to a pipeline, so a new word could be clearer

File: doc/docs/master/concepts/data-concepts/file.md [hunk 2]
```diff
@@ -21,12 +21,12 @@ Pachyderm provides the following file processing strategies:
 
 **Appending files**
 :   By default, when you put a file into a Pachyderm repository and a
-    file by the same name already exists in the repo, Pachyderm appends
-    the new data to the existing file.
+    file by the same name already exists in the repo, Pachyderm overwrites
+    the existing file with the new data.
```
Comment (PFedak) [inline doc/docs/master/concepts/data-concepts/file.md:25]: (default flipped)

File: doc/docs/master/concepts/data-concepts/index.md [hunk 3]
```diff
@@ -25,22 +25,26 @@ run into a merge conflict with Pachyderm.
...
-:   A Pachyderm repository is the highest level data object. Typically,
-    each dataset in Pachyderm is its own repository.
+:   A Pachyderm repository is the highest level data object,
+    an independent file system. Typically, each dataset in
+    Pachyderm is its own repository.
 
 **Commit**
-:   A commit is an immutable snapshot of a repo at a particular point
-    in time.
+:   A commit is an immutable snapshot of a data in Pachyderm corresponding
```
Comment (PFedak) [inline doc/docs/master/concepts/data-concepts/index.md:33]: feels superfluous to have "sub-commit" as a top level entry here, but maybe that helps call it out more clearly?

File: doc/docs/master/concepts/data-concepts/provenance.md [hunk 4]
```diff
@@ -45,7 +45,8 @@ complete audit trail so that all results are reproducible.
 
 The following diagram demonstrates how provenance works:
 
-![Provenance example](../../assets/images/provenance.svg)
+![Provenance example](../../assets/images/provenance.svg) (needs updated commit IDs)
```
[the remark this answers is not in the document]
Comment (nadegepepin) [reply doc/docs/master/concepts/data-concepts/provenance.md:48]: I created a separate Issue to come back to this: #6730

File: doc/docs/master/concepts/data-concepts/repo.md (hunk 1 above)
Comment (nadegepepin) [reply doc/docs/master/concepts/data-concepts/repo.md:40]: Removed the term `Source`. Replaced with `Input repo with no provenance`.

File: doc/docs/master/concepts/data-concepts/index.md (hunk 3 above)
Comment (nadegepepin) [reply doc/docs/master/concepts/data-concepts/index.md:33]: Added the following issue to come back to that definition: #6760

**Review by nadegepepin — APPROVED** [review]
LGTM -  I believe that the main concepts are in and need to be published ASAP -  I created a couple of GH issues to come back to what still needs some work (see #6760, for example). A couple of precisions, a global work on consistency, renaming sub-commit->commit, etc...

**Pull request merged**